### PR TITLE
Use g_want_determinism more

### PR DESCRIPTION
**Pull request by JosJuice**
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3979)

### CODE REVIEW

**Comment by JosJuice** [comment]
The first commit might cause problems if [issue 9681](https://bugs.dolphin-emu.org/issues/9681) isn't fixed first.

File: Source/Core/Core/Core.cpp [hunk 1]
```diff
@@ -976,8 +976,7 @@ void UpdateWantDeterminism(bool initial)
   // For now, this value is not itself configurable.  Instead, individual
   // settings that depend on it, such as GPU determinism mode. should have
   // override options for testing,
-  bool new_want_determinism =
-      Movie::IsPlayingInput() || Movie::IsRecordingInput() || NetPlay::IsNetPlayRunning();
```
Comment (magcius) [inline Source/Core/Core/Core.cpp]: why doesn't playing input want determinism?

Comment (magcius) [inline Source/Core/Core/Core.cpp]: oh, nvm, you swapped it with IsMovieActive which checks both cases

**Comment by Parlane** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3979)*

**Pull request merged**